app/src/main/java/com/poklad/androidifystore/data/repositories/ProductsRemoteRepositoryImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.poklad.androidifystore.data.repositories
+
+import com.poklad.androidifystore.data.mapper.ProductResponseToProductItemMapper
+import com.poklad.androidifystore.data.remote.StoreApi
+import com.poklad.androidifystore.domain.model.ProductItem
+import com.poklad.androidifystore.domain.repositories.ProductsRepository
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
not sure that you need singleton here. I'd rather recommend you to read about Dagger's `Scopes` and define your scopes. It's general note for your Dagger setup, as you are using dagger, but you doesn't use Dagger's main feature